saleor/product/filters.py

```diff
@@ -24,15 +22,11 @@
 
 
 def filter_products_by_attributes_values(qs, queries: T_PRODUCT_FILTER_QUERIES):
-    # Combine filters of the same attribute with OR operator
-    # and then combine full query with AND operator.
-    combine_and = [
-        Q(**{"attributes__values__pk__in": values_pk})
-        | Q(**{"variants__attributes__values__pk__in": values_pk})
-        for _, values_pk in queries.items()
-    ]
-    query = functools.reduce(operator.and_, combine_and)
-    qs = qs.filter(query).distinct()
```

Comment:
Are you sure that we don't need `distinct` here? I remember that I had some problems with duplicated rows.

Reply:
I was testing a couple of cases and none of them returned duplicated rows

Reply:
It will return duplicate rows. A better implementation would use an `Exists` subquery together with `OuterRef` instead of filtering on a many-to-many relation (which would also eliminate the need to use `distinct`).